app/two_factor_authentication/use_cases/create_new_user_2fa_use_case.py

```diff
@@ -0,0 +1,54 @@
+from pydantic import EmailStr
+from sqlalchemy.orm import Session
+import pyotp
+from app.common.exceptions.model_not_created_exception import (
+    ModelNotCreatedException,
+)
+from app.common.exceptions.model_not_found_exception import (
+    ModelNotFoundException,
+)
+from app.core.config import get_settings
+from app.two_factor_authentication.repositories.user_2fa_repository import (
+    users_2fa_repository,
+)
+from app.two_factor_authentication.schemas.user_2fa_schema import (
+    User2FACreate,
+    User2FAResponse,
+)
+from app.two_factor_authentication.services.user_2fa_service import (
+    Users2FAService,
+)
+from app.users.repositories.users_repository import users_repository
+from app.users.services.users_service import UsersService
+
+settings = get_settings()
+
+
+class CreateNewUser2FAUseCase:
+    def __init__(self, session: Session):
+        self.session = session
```

Comment:
I would initialize services in the `__init__` method to encapsulate all object initialization there, while keeping the namespace of the `execute` method and keeping it more streamlined.

```suggestion
    def __init__(self, session: Session):
        self.session = session
        self.users_service = UsersService(session, users_repository)
        self.users_2fa_service = Users2FAService(session, users_2fa_repository)
```